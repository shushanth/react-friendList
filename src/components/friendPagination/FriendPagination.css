:local(.friendPagination) {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: center;
  padding: 1em;
  width: 100%;
  overflow-x: auto;
}
