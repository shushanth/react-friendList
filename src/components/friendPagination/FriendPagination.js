import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

import { Pagination } from "../../shared/components";
import styles from "./FriendPagination.css";

const FriendPagination = ({
  selectedPage,
  pageSize,
  totalPages,
  onPageChange
}) => {
  return (
    <div className={classnames(styles.friendPagination)}>
      <Pagination
        pageSize={pageSize}
        totalPages={totalPages}
        pageChange={onPageChange}
        selectedPage={selectedPage}
      />
    </div>
  );
};

FriendPagination.PropTypes = {
  pageSize: PropTypes.number,
  totalPages: PropTypes.number,
  selectedPage: PropTypes.number,
  pageChange: PropTypes.func
};

export default FriendPagination;
