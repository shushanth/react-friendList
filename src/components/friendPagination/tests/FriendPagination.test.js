import React from "react";
import { mount, shallow } from "enzyme";

import FriendPagination from "../FriendPagination";

describe("Component: <FriendPagination/>", () => {
  let props;
  let FriendPaginationWrapper;
  let FriendPaginationMountWrapper;
  let getTotalPageBtnCount;
  let getButtonTextValues;
  beforeAll(() => {
    props = {
      pageSize: 2,
      totalPages: 5,
      pageChange: jest.fn(),
      selectedPage: 1
    };
    FriendPaginationWrapper = shallow(<FriendPagination {...props} />);
    FriendPaginationMountWrapper = mount(<FriendPagination {...props} />);
    getTotalPageBtnCount = value => Math.ceil(value / props.pageSize);
    getButtonTextValues = position => {
      return parseInt(
        FriendPaginationMountWrapper.find("Pagination button")
          .at(position)
          .text(),
        10
      );
    };
  });

  it("should render without crashing", () => {
    expect(FriendPaginationWrapper.exists()).toBe(true);
    expect(FriendPaginationWrapper).toMatchSnapshot();
  });

  it("should match the totalPages and pagination buttons count", () => {
    expect(
      FriendPaginationMountWrapper.find("Pagination button").length
    ).toEqual(getTotalPageBtnCount(5));
  });

  it("should have proper button values", () => {
    expect(getButtonTextValues(0)).toEqual(1);
    expect(getButtonTextValues(1)).toEqual(2);
    expect(getButtonTextValues(2)).toEqual(3);
  });

  it("should have pagination buttons count based on the total pages count", () => {
    let friendPagination = mount(
      <FriendPagination {...{ totalPages: 3, pageSize: 2 }} />
    );
    let friendPaginationMore = mount(
      <FriendPagination {...{ totalPages: 10, pageSize: 2 }} />
    );
    expect(friendPagination.find("Pagination button").length).toEqual(
      getTotalPageBtnCount(3)
    );
    expect(friendPaginationMore.find("Pagination button").length).toEqual(
      getTotalPageBtnCount(10)
    );
  });
});
