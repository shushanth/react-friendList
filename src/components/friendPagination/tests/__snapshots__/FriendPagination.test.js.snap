exports[`Component: <FriendPagination/> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FriendPagination
    pageChange={[Function]}
    pageSize={2}
    selectedPage={1}
    totalPages={5} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Pagination
        pageChange={undefined}
        pageSize={2}
        selectedPage={1}
        totalPages={5} />,
      "className": "",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "pageChange": undefined,
        "pageSize": 2,
        "selectedPage": 1,
        "totalPages": 5,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Pagination
          pageChange={undefined}
          pageSize={2}
          selectedPage={1}
          totalPages={5} />,
        "className": "",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "pageChange": undefined,
          "pageSize": 2,
          "selectedPage": 1,
          "totalPages": 5,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
